x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/openai/OpenAiUnifiedStreamingProcessor.java

```diff
@@ -71,7 +79,25 @@ protected void upstreamRequest(long n) {
     @Override
     protected void next(Deque<ServerSentEvent> item) throws Exception {
         var parserConfig = XContentParserConfiguration.EMPTY.withDeprecationHandler(LoggingDeprecationHandler.INSTANCE);
-        var results = parseEvent(item, OpenAiUnifiedStreamingProcessor::parse, parserConfig, logger);
+
+        var results = new ArrayDeque<StreamingUnifiedChatCompletionResults.ChatCompletionChunk>(item.size());
+        for (var event : item) {
+            if (ServerSentEventField.EVENT == event.name() && "error".equals(event.value())) {
+                previousEventWasNotError = false;
```

Comment:
nit: This might be a bit easier to read using the positive form `previousEventWasAnError = true`.

Then in the check below we could do:

```
if (previousEventWasAnError) {
  throw errorParser.apply(event.value(), null);
}

try {

} catch () {

}

```